### ISSUE TITLE
Merge pystiche.ops into pystiche.loss

### ISSUE THREAD

**Issue by pmeier** [pystiche/pystiche#436]
Right now the only functional difference between an `pystiche.ops.Operator` and `pystiche.loss.MultiOperatorLoss` is that the latter properly handles the multi-layer encoding:

https://github.com/pystiche/pystiche/blob/ad9b3ea09f9803249b162fbe009db25869e42fb4/pystiche/loss/multi_op.py#L31-L35

https://github.com/pystiche/pystiche/blob/ad9b3ea09f9803249b162fbe009db25869e42fb4/pystiche/loss/multi_op.py#L60-L71

Thus, if the user needs the multi-layer encoding paradigm, he needs to remember to wrap it in a `pystiche.loss.MultiOperatorLoss` or any of the subclasses. This is especially unintuitive if a single `ops.MultiLayerEncodingOperator` is used:

```python
from pystiche import enc, ops, loss

texture_synthesis_loss = ops.MultiLayerEncodingOperator(
    enc.vgg19_multi_layer_encoder(),
    ("relu1_1", "relu2_1", "relu3_1", "relu4_1", "relu5_1"),
    lambda encoder, layer_weight: ops.GramOperator(encoder, score_weight=layer_weight),
)
criterion = loss.MultiOperatorLoss([("texture_synthesis_loss", texture_synthesis_loss)])
```

---

To overcome this I propose to split the handling of the multi-layer encoding into a separate class:

```python
class MultiLayerEncoderHandler:
    def __init__(self, criterion, trim=True):
        self.multi_layer_encoders = self.collect_multi_layer_encoders()
        if trim:
            self.trim()

    def trim(self):
        for multi_layer_encoder in self.multi_layer_encoders:
            multi_layer_encoder.trim()

    def clear_storage(self):
        for multi_layer_encoder in self.multi_layer_encoders:
            multi_layer_encoder.clear_storage()
```
This does not need to handle the encoding after #435 is implemented. 
By using a `MultiLayerEncoderHandler`
With this an optimization loop might look like this:

```python

mle_handler = MultiLayerEncoderHandler(criterion, trim=trim)

for step in range(num_steps):

    def closure():
        return criterion(input)

    optimizer.step(closure)
    mle_handler.clear_storage()
```

Up to this point this change is BC. 

---

The change gets BC breaking if we merge `PerceptualLoss` and `GuidedPerceptualLoss` with `pystiche.ops` and remove `pystiche.loss.MultiOperatorLoss` since it is now obsolete.

As a last proposal I favor to merge `pystiche.ops` into `pystiche.loss`, i.e. removing `pystiche.ops` to make the purpose more clear.

**Comment by pmeier** [comment]
Progress is tracked on branch [`dev/merge-ops-loss`](https://github.com/pystiche/pystiche/tree/dev/merge-ops-loss)
